rbc/irtools.py

```diff
@@ -287,6 +289,12 @@ def compile_instance(func, sig,
         flags.set('no_cpython_wrapper')
         flags.set('no_cfunc_wrapper')
 
+    # set CAPTURED_ERROS to "new_style"
+    # https://github.com/numba/numba/pull/7397
+    os.environ["CAPTURED_ERRORS"] = "new_style"
+    if 'numba' not in sys.modules:
```

Comment:
This does not make sense. When the code execution hits this, numba is surely imported.

The `reload_config` call should happen at the early stage of loading `rbc` modules, say, in `rbc/__init__.py` (so that we do it only once) and only if `numba` has been imported before `rbc` and after resetting CAPTURED_ERRORS env variable (otherwise there is nothing new to reload).

In addition, when a user specifies `CAPTURED_ERRORS` env variable, we should respect that:
```python
if "CAPTURED_ERRORS" not in os.environ:
    os.environ["CAPTURED_ERRORS"] = "new_style"
```

Reply:
Would it be ok if we keep the `config.reload_config()` here? It's a cheap operation that will not slow down RBC.

Reply:
I could see it being cheap but reading this code can be confusing: each call to a function sets environment variable and reloads config, and one may wonder if this is intentional and why.

I would be ok with:
```
if "CAPTURED_ERRORS" not in os.environ:
    os.environ["CAPTURED_ERRORS"] = "new_style"
    config.reload_config()
```
that ensures that the env set and reload_config is called exactly once (unless users explicitly do `del os.environ["CAPTURED_ERRORS"]`, for instance) plus if you would be ok if at some point this code is moved elsewhere :)